### DISCUSSION TITLE
Order vs Sort in C#

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#156] [Q&A] [answered]
Happy New Year.

What are the differences between Sort and Order in C#? When and Where should we use these?
The result for the below example is the same in Sort and Order.

```
  var listInt = new List<int> { 5, 8, 9, 17, -6, 55 };
        var stuOrder = listInt.Order();
        listInt.Sort();
        Console.WriteLine(string.Join(',', stuOrder));
        Console.WriteLine(string.Join(',', listInt));
```

**Comment by christiannagel** [comment, marked as answer]
`Order` is a LINQ extension method from the Enumerable class available since .NET 7. Because this method extends `IEnumerable<T>`, you can use the `Order` method with any collection class implementing `IEnumerable<T>`. The `Order` method returns a new collection and doesn't change the source.

The `Sort` method is defined with `List<T>`. This method is optimized for the`List<T>` type and changes the sort order of the originating collection. 

The `Array` type implements a `Sort` method as well.

That said, use the `Order` method with any collection type, when you just have `IEnumerable<T>` available. If you use the List<T> type sorted, use the `Sort` method if you want to have the original collection sorted. In case you need this sort order independent of the original collection, use the `Order` method as well.

**Reply by ShervanN** [reply]
Thank you
